src/Data/Mapper/PositionMapper.php

```diff
@@ -27,4 +27,24 @@ public function getPositions($options = [])
 
         return [$this->mapObjects($result), $paginate];
     }
+
+    protected function mapObjects(array $records)
+    {
+        $results = [];
+        foreach ($records as $record) {
+            $entity = new $this->entity;
+            $entity->setId($record['id']);
+            $entity->setAccountId($record['account_id']);
+            $entity->setName($record['name']);
+            $entity->setColor($record['color']);
+            $entity->setSort($record['sort']);
+            $entity->setCreatedAt($record['created_at']);
+            $entity->setUpdatedAt($record['updated_at']);
+            $entity->setDeleted($record['is_deleted']);
```

Comment:
If the intent is to be completely verbose, the entity `__construct` would look like:

```php
// class Entity\Position
public function __construct(
    $id,
    $account_id,
    $name,
    $color,
    ...
) { ... }
```

I personally don't like this approach because it is overly verbose. Ideally, only the factory and entity signatures change, but even a small mistake there can cause issues.